salt/modules/timezone.py

```diff
@@ -18,14 +18,22 @@
 
 log = logging.getLogger(__name__)
 
+__virtualname__ = 'timezone'
+
 
 def __virtual__():
     '''
     Only work on POSIX-like systems
     '''
     if salt.utils.is_windows():
-        return (False, 'The timezone execution module failed to load: not available on Windows systems.')
-    return True
+        return (False, 'The timezone execution module failed to load: '
+                       'not available on Windows systems.')
+
+    if salt.utils.is_darwin():
+        return (False, 'The timezone execution module failed to load: '
+                       'not available on MacOS systems.')
```

Comment:
Hmmm, I can actually see both points. I guess from a user perspective, if for some reason the mac_timezone module didn't load, they would probably get this error message. If I'm a new user, that leads me to believe that there's no timezone support on mac, when it really is just a problem with the mac support (that does exist). So I think I'm with @cachedout, we should probably point them to that module so they can investigate why it didn't load.

Maybe something like this: "The timezone execution module failed to load: mac_timezone.py should be replace this module on OS X. There was a problem loading that module."

Clunky, but gives the user more useful information.